controllers/metal3.io/hostfirmwarecomponents_controller.go

```diff
@@ -196,35 +191,43 @@ func (r *HostFirmwareComponentsReconciler) updateHostFirmware(info *rhfcInfo, co
 	reason := reasonValidComponent
 	generation := info.hfc.GetGeneration()
 
+	newStatus := info.hfc.Status.DeepCopy()
+	// change the Components in Status
+	newStatus.Components = make([]metal3api.FirmwareComponentStatus, len(components))
+	for i := range info.hfc.Status.Components {
+		components[i].DeepCopyInto(&newStatus.Components[i])
+	}
```

Comment:
I've changed  to this approach to see if would help, but maybe it's missing to trigger a condition before doing this, checking if `!reflect.DeepEqual(info.hfc.Status.Components, components)` ?